src/eei.cpp

```diff
@@ -659,6 +657,40 @@ namespace hera {
         dataLength = static_cast<uint32_t>(arguments[3].geti32());
       }
 
+      return Literal(eeiCall(kind, gas, addressOffset, valueOffset, dataOffset, dataLength));
+    }
+
+    if (import->base == Name("create")) {
+      heraAssert(arguments.size() == 4, string("Argument count mismatch in: ") + import->base.str);
+
+      uint32_t valueOffset = static_cast<uint32_t>(arguments[0].geti32());
+      uint32_t dataOffset = static_cast<uint32_t>(arguments[1].geti32());
+      uint32_t length = static_cast<uint32_t>(arguments[2].geti32());
+      uint32_t resultOffset = static_cast<uint32_t>(arguments[3].geti32());
+
+      return Literal(eeiCreate(valueOffset, dataOffset, length, resultOffset));
+    }
+
+    if (import->base == Name("selfDestruct")) {
+      heraAssert(arguments.size() == 1, string("Argument count mismatch in: ") + import->base.str);
+
+      uint32_t addressOffset = static_cast<uint32_t>(arguments[0].geti32());
+      eeiSelfDestruct(addressOffset);
+
+      return Literal();
+    }
+
+    heraAssert(false, string("Unsupported import called: ") + import->module.str + "::" + import->base.str + " (" + to_string(arguments.size()) + "arguments)");
+  }
+
+  void EthereumInterface::eeiReturnDataCopy(uint32_t dataOffset, uint32_t offset, uint32_t size)
+  {
+      safeChargeDataCopy(size, GasSchedule::verylow);
+      storeMemory(m_lastReturnData, offset, dataOffset, size);
+  }
+
+  uint32_t EthereumInterface::eeiCall(EEICallKind kind, int64_t gas, uint32_t addressOffset, uint32_t valueOffset, uint32_t dataOffset, uint32_t dataLength)
```

Comment:
I think it may make sense to trade some code repetition for more clarity here, as in #148 where they are split.

Reply:
That was the main point of enum so that it can be shared 😉